### PR TITLE
Code integration for December 9th kuka demo 

### PR DESCRIPTION
**Pull request by YeZhao**
A few comments/instructions for running dircol and torque control of a grasping demo.

-- Make sure the robot is in near 0 position configuration (i.e., straight up position, using either position control mode or jogging). Launch DrakeFRITorqueDriver
(currently, once the torque controller is executed on the robot, no worries about this low-level controller)

-- Run the following command to launch the procman in kuka-dev/drake directory
build/install/bin/bot-procman-sheriff -l ./drake/examples/kuka_iiwa_arm/kuka_planning_and_control.pmd. 
Most components in the system diagram block (drawn by Scott) have been incorporated. Users can choose to run either dircol or ik planners as needed in GUI.

-- Follow the normal routine to launch the director, make plans and execute (the dircol trajectory generation may take 10-20 seconds). Click gripper buttons to control its close and open. 

-- If users run dircol for trajectory generation, matlab should be launched. Instructions can be seen from the instructions in the previous pull request https://github.com/HarvardAgileRoboticsLab/drake/pull/10.
 
-- Right now, the new urdf with the gripper is drake/examples/kuka_iiwa_arm/urdf/iiwa14_fixed_gripper.urdf. It includes both arm and gripper. It is called by kuka_plan_runner.cc, kuka_control.cc, iiwa_simulation.cc, director_config.json (for director visualization). 

-- The gripper compensation is implemented and improved in kuka_control. Right now, we have a signal RigidBodyTree (use one urdf, iiwa14_fixed_gripper.urdf). Kuka low-level controllers do not provide source files so it seems difficult to modify gravity comp in the control cabinet (there might be some chance to tune the model via kuka APIs, but the chance seems quite less according to comments from Sam and Liang).

### CODE REVIEW

File: drake/examples/kuka_iiwa_arm/kuka_lcm_planner.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ class KukaIkPlanner : public KukaPlanner{
       // unpack the LCM message
       auto constraints = parse_constraints(status->constraints, kuka_);
       std::vector<RigidBodyConstraint*> constraint_ptrs(constraints.size());
-      for (int i=0; i<constraints.size(); i++){
+      for (unsigned int i=0; i<constraints.size(); i++){
```
Comment (kuindersma) [inline drake/examples/kuka_iiwa_arm/kuka_lcm_planner.h:145]: why was this change necessary?

**Comment by kuindersma** [comment]
Sorry, looks like this one has a merge conflict now that I've pulled in Patrick's branch

**Comment by YeZhao** [comment]
@kuindersma Here is the cleaned-up code for December 9th demo. The merge conflict you mentioned before should be solved now. New changes are updated in the instructions/comments at the beginning of this page.

File: drake/examples/kuka_iiwa_arm/KukaPlanner.m [hunk 2]
```diff
@@ -71,8 +74,8 @@ function handle_request(obj, msg)
             
             % joint limits
             [jl_min, jl_max] = obj.kuka.getJointLimits;
-            x_min = [jl_min; -inf(Nq,1)];
-            x_max = [jl_max;  inf(Nq,1)];
+            x_min = [jl_min; -.75*ones(Nv,1)];
+            x_max = [jl_max;  .75*ones(Nv,1)];
```
Comment (kuindersma) [inline drake/examples/kuka_iiwa_arm/KukaPlanner.m]: Could we turn .75 into a named parameter like "max_joint_velocity" or similar to make it more readable?

File: drake/examples/kuka_iiwa_arm/KukaPlanner.m [hunk 3]
```diff
@@ -96,11 +99,11 @@ function handle_request(obj, msg)
...
-            % x
-            % info
-            % traj = xtraj.setOutputFrame(obj.kuka.getStateFrame());
-            % v = obj.kuka.constructVisualizer();
-            % playback(v,traj,struct('slider',true));
+            x
+            info
+            traj = xtraj.setOutputFrame(obj.kuka.getStateFrame());
+            v = obj.kuka.constructVisualizer();
+            playback(v,traj,struct('slider',true));
```
Comment (kuindersma) [inline drake/examples/kuka_iiwa_arm/KukaPlanner.m]: I think this should be commented out

File: drake/matlab/solvers/NonlinearProgram.m [hunk 4]
```diff
@@ -161,9 +161,9 @@
       obj.solver_options.snopt.MajorIterationsLimit = 1000;
       obj.solver_options.snopt.MinorIterationsLimit = 500;
       obj.solver_options.snopt.IterationsLimit = 10000;
-      obj.solver_options.snopt.MajorOptimalityTolerance = 1e-6;
-      obj.solver_options.snopt.MajorFeasibilityTolerance = 1e-6;
-      obj.solver_options.snopt.MinorFeasibilityTolerance = 1e-6;
+      obj.solver_options.snopt.MajorOptimalityTolerance = 1e-3;
+      obj.solver_options.snopt.MajorFeasibilityTolerance = 1e-3;
+      obj.solver_options.snopt.MinorFeasibilityTolerance = 1e-3;
       obj.solver_options.snopt.SuperbasicsLimit = 300;
```
Comment (kuindersma) [inline drake/matlab/solvers/NonlinearProgram.m:167]: This shouldn't be changed

**Comment by YeZhao** [comment]
All the three comments have been addressed.
